src/Microsoft.Health.Fhir.Core/Features/Search/Expressions/Parsers/ExpressionParser.cs

```diff
@@ -101,7 +101,7 @@ public IncludeExpression ParseInclude(string resourceType, string includeValue,
                 refSearchParameter = _searchParameterDefinitionManager.GetSearchParameter(originalType.ToString(), searchParam.ToString());
             }
 
-            return new IncludeExpression(resourceType, refSearchParameter, targetType, wildCard, isReversed);
+            return new IncludeExpression(resourceType, refSearchParameter, originalType.ToString(), targetType, wildCard, isReversed);
```

Comment:
This will help with addressing https://github.com/microsoft/fhir-server/issues/1236